data-otservbr-global/npc/emperor_kruzak.lua

```diff
@@ -62,37 +62,32 @@ local function creatureSayCallback(npc, creature, type, message)
 		npcHandler:say("In exchange for a truly generous donation, I will offer a special outfit. Do you want to make a donation?", npc, creature)
 		npcHandler:setTopic(playerId, 1)
 	elseif MsgContains(message, "yes") then
-		-- vamos tratar todas condições para YES aqui
 		if npcHandler:getTopic(playerId) == 1 then
-			-- para o primeiro Yes, o npc deve explicar como obter o outfit
 			npcHandler:say({
 				"Excellent! Now, let me explain. If you donate 1.000.000.000 gold pieces, you will be entitled to wear a unique outfit. ...",
 				"You will be entitled to wear the {armor} for 500.000.000 gold pieces, {helmet} for an additional 250.000.000 and the {boots} for another 250.000.000 gold pieces. ...",
 				"What will it be?",
 			}, npc, creature)
 			npcHandler:setTopic(playerId, 2)
-			-- O NPC só vai oferecer os addons se o player já tiver escolhido.
 		elseif npcHandler:getTopic(playerId) == 2 then
-			-- caso o player repita o yes, resetamos o tópico para começar de novo?
 			npcHandler:say("In that case, return to me once you made up your mind.", npc, creature)
 			npcHandler:setTopic(playerId, 0)
-			-- Inicio do outfit
-		elseif npcHandler:getTopic(playerId) == 3 then -- ARMOR/OUTFIT
-			if player:getStorageValue(Storage.OutfitQuest.GoldenOutfit) < 1 then
-				if player:getMoney() + player:getBankBalance() >= 500000000 then
+		elseif npcHandler:getTopic(playerId) == 3 then
+			local goldenOutfitQuest = player:kv():get("golden-outfit-quest") or 0
```

Comment:
you can use this variable at the start of the script to avoid checking many times.